### ISSUE TITLE
Turn rusqlite's `unlock_notify` feature on in crates that use rusqlite.

### ISSUE THREAD

**Issue by thomcc** [mozilla/application-services#3287]
This will bring us into par with how [mozStorage](https://searchfox.org/mozilla-central/source/storage/mozStorageConnection.cpp#1084-1099) performs sqlite3_step, and has a chance to reduce the frequent SQLITE_BUSY errors (https://github.com/mozilla/application-services/issues/2873) we see. However, I would not count on this, but even if it does not cause this, and a more convincing reason to do it is that it should improve latency when one connection holds a lock that another connection is waiting on.

Note that given our mildly heavy use of execute_batch we may want to wait for the next rusqlite release (a week or two) to do this, as execute_batch will not participate in unlock notifications as prior to recently they went through sqlite3_exec and not sqlite3_step. That said, there's no real harm in turning it on before then, it just won't do as much good. (I'm filing this now mostly because the code just landed in rusqlite for this, which reminded me, although I've thought we should turn this on for a while now)

This is a relatively trivial change aside from that, and just requires adding a cargo feature to the crates that use rusqlite.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/SYNC-1429)
┆Epic: important not urgent
┆Sprint: SYNC - end 2020-07-17

**Comment by thomcc** [comment]
Thinking more, I think it actually could reduce SQLITE_BUSY errors, since this will cause connections to give up locks sooner than they otherwise would.
